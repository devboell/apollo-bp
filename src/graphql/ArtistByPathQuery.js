import gql from 'graphql-tag'

export default gql`
  query ArtistByPathQuery($path: String) {
    artistByPath(path: $path) {
      path
      paintings {
        title
        name
        genre
      }
    }
  }
`
