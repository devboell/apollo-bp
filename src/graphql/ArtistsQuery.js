import gql from 'graphql-tag'

export default gql`
  query ArtistsQuery {
    artists {
      name
      path
    }
  }
`
