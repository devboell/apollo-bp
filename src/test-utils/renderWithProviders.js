import React from 'react'

import { render } from 'react-testing-library'
import { compose } from 'ramda'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history' /* eslint-disable-line */
import { MockedProvider } from 'react-apollo/test-utils'
import { Provider } from 'react-redux'


const withRouter = ([node, config]) => {
  const history = createMemoryHistory({ initialEntries: [config.route] })
  return [
    <Router history={history}>{node}</Router>,
    { ...config, history },
  ]
}

const withApollo = ([node, config]) => [
  <MockedProvider
    removeTypename
    mocks={config.mocks}
  >
    {node}
  </MockedProvider>,
  config,
]

const withRedux = ([node, config]) => [
  <Provider store={config.store}>{node}</Provider>,
  config,
]

const withProvider = key => fn => ([node, config]) => (
  !config[key]
    ? [node, config]
    : fn([node, config])
)

const withProviders = compose(
  withProvider('route')(withRouter),
  withProvider('mocks')(withApollo),
  withProvider('store')(withRedux),
)

const renderWithProviders = (node, config) => {
  const [WithProviders, { history }] = withProviders([node, config])
  return {
    ...render(WithProviders),
    history,
  }
}


// re-export everything
export * from 'react-testing-library'

// override render method
export { renderWithProviders }
