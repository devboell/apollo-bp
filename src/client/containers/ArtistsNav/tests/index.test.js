import React from 'react'
import { print } from 'graphql'
import { MockedProvider } from 'react-apollo/lib/test-utils'

import artistsData from 'data/test/artistsQuery.json'

import ArtistsNavWithData, {
  ArtistsNav,
  ARTISTS_QUERY,
  withArtists,
 } from '../index'


describe('ArtistsNav component', () => {
  it('with data snapshot', () => {
    const wrapper = shallow(
      <ArtistsNav artists={artistsData} loading={false} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('without data snapshot', () => {
    const wrapper = shallow(
      <ArtistsNav artists={[]} loading />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('ArtistsNav ARTISTS_QUERY', () => {
  it('has a known query shape', () => {
    expect(print(ARTISTS_QUERY)).toMatchSnapshot()
  })
})

/* eslint-disable react/prop-types */
describe('ArtistsNav withArtists', () => {
  it('renders with loading first', () => {
    class Container extends React.Component {
      componentWillMount() {
        expect(this.props.loading).toBe(true)
        expect(this.props.artists).toBeFalsy()
      }
      render() {
        return null
      }
    }
    const ContainerWithData = withArtists(Container)
    render(
      <MockedProvider
        mocks={[
          { request: { ARTISTS_QUERY }, result: { data: artistsData } },
        ]}
      >
        <ContainerWithData />
      </MockedProvider>,
    )
  })

  /* eslint-disable react/no-multi-comp */
  it('renders data without crashing', () => {
    class Container extends React.Component {
      componentWillReceiveProps(nextProps) {
        expect(nextProps.loading).toBe(false)
        expect(nextProps.artists).toEqual(artistsData)
      }
      render() {
        return null
      }
    }
    const ContainerWithData = withArtists(Container)
    render(
      <MockedProvider
        mocks={[
          { request: { ARTISTS_QUERY }, result: { data: artistsData } },
        ]}
      >
        <ContainerWithData />
      </MockedProvider>,
    )
  })
})

describe('ArtistsNav, enhanced component', () => {
  it('renders without crashing', () => {
    const wrapper = render(
      <MockedProvider
        mocks={[
          { request: { ARTISTS_QUERY }, result: { data: artistsData } },
        ]}
      >
        <ArtistsNavWithData />
      </MockedProvider>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
