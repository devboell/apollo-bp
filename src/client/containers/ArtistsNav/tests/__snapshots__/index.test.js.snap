// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsNav ARTISTS_QUERY has a known query shape 1`] = `
"query ArtistsQuery {
  artists {
    name
    path
  }
}
"
`;

exports[`ArtistsNav component with data snapshot 1`] = `
<styled.div>
  <styled.ul>
    <ArtistNavItem
      name="Heroshige"
      path="heroshige"
    />
    <ArtistNavItem
      name="Hokusai"
      path="hokusai"
    />
    <ArtistNavItem
      name="Utamaro"
      path="utamaro"
    />
    <ArtistNavItem
      name="Kunisada"
      path="kunisada"
    />
  </styled.ul>
</styled.div>
`;

exports[`ArtistsNav component without data snapshot 1`] = `
<div>
  loading...
</div>
`;

exports[`ArtistsNav, enhanced component renders without crashing 1`] = `
<div>
  loading...
</div>
`;
